### PR TITLE
Legacy fullscreen

### LINKED ISSUE #709: Something wrong when using an external displayer
I connect an external displayer to my macbook.When I use iina with full screen mode at one of my displayer, another one would displayer nothing but a black screen.

### LINKED ISSUE #909: [Feature Request] Set default display
An option to select the display on which all files are opened.

**System and IINA version:**

- macOS 10.12.6
- IINA 0.0.12 (build 37)

**Expected behavior:**

An option to set the default display on which all files are opened would be great for multi-display users. If you have to open a bunch of files using the main screen, but want to view them on a secondary screen, you have to drag them over.

Setting a default display, or remember the last used display, would be a great addition.

### PR DESCRIPTION
**Pull request by lhc70000**
**Description:**

This PR will add legacy full screen support and fix #709, fix #909.

### CODE REVIEW

File: iina/MainWindowController.swift [hunk 1]
```diff
@@ -1247,6 +1251,7 @@ class MainWindowController: NSWindowController, NSWindowDelegate {
   }
 
   func windowDidBecomeMain(_ notification: Notification) {
+    print("became main")
```
Comment (saagarjha) [inline iina/MainWindowController.swift]: Might want to remove this

File: iina/AppData.swift [hunk 2]
```diff
@@ -93,6 +93,7 @@ struct Constants {
     static let keyBindingInputChanged = Notification.Name("IINAkeyBindingInputChanged")
     static let fileLoaded = Notification.Name("IINAFileLoaded")
     static let historyUpdated = Notification.Name("IINAHistoryUpdated")
+    static let legacyFullScreen = Notification.Name("IINALegacyFullScreen")
```
Comment (saagarjha) [inline iina/AppData.swift]: Not completely related, but I think this struct would be better if it was an extension of `Notification.Name`.

File: iina/MainWindowController.swift [hunk 3]
```diff
@@ -1836,7 +1841,70 @@ class MainWindowController: NSWindowController, NSWindowDelegate {
...
+        window.styleMask.insert(.borderless)
+        window.styleMask.insert(.fullScreen)
+        // cancel aspect ratio
+        window.resizeIncrements = NSSize(width: 1, height: 1)
+        // auto hide menubar and dock
+        NSApp.presentationOptions.insert(.autoHideMenuBar)
+        NSApp.presentationOptions.insert(.autoHideDock)
+        // set frame
+        let screen = window.screen ?? NSScreen.main()!
+        window.setFrame(NSRect(origin: NSZeroPoint, size: screen.frame.size), display: true, animate: true)
```
Comment (saagarjha) [inline iina/MainWindowController.swift]: Does `.zero` work here? I think `NSZeroPoint` needs to be updated, but you should be able to use `CGPoint.zero` while you wait.

File: iina/AppData.swift (hunk 2 above)
Comment (lhc70000) [inline iina/AppData.swift]: Yep, I was exactly about to do the same thing. Maybe in the next refactor branch.

**Pull request merged**